integration-tests/datasource/src/test/java/io/quarkus/it/datasource/ValidationQueryTest.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.it.datasource;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import jakarta.inject.Inject;
+
+import org.h2.jdbc.JdbcSQLNonTransientConnectionException;
+import org.h2.tools.Server;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class ValidationQueryTest {
+
+    @BeforeEach
+    public void setup() throws SQLException {
+        startH2DB();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        stopH2DB();
+    }
+
+    //tag::injection[]
+    @Inject
+    AgroalDataSource defaultDataSource;
+    //end::injection[]
+
+    private Server tcpServer;
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
```

Comment:
`QuarkusUnitTest` is not intended to be used this way AFAIK; it's more about extension tests, within a `deployment` module.

In any case... you can probably avoid creating the new `datasource` module altogether -- which would require more work than what's in this PR, due to how our CI works -- and move this new test to `extensions/agroal/deployment/src/test/java` instead?

Reply:
At first, I tried placing it under `extensions/datasource/deployment/src/test/java`, but it didn’t work due to a circular dependency caused by the need to depend on agroal. 
So I moved it to `integration-tests`, though I did feel it was a bit out of place.
If it's okay to put it in `extensions/agroal/deployment/src/test/java`, I'll move it there.

Reply:
>  If it's okay to put it in extensions/agroal/deployment/src/test/java, I'll move it there.

It's definitely okay to do this!